components/scream/src/share/io/scorpio_output.cpp

```diff
@@ -1,102 +1,143 @@
-
-  // Create map of fields in this output with the field_identifier in the field manager.
-  auto& var_params = m_params.sublist("FIELDS");
-  for (int var_i=0; var_i<var_params.get<Int>("Number of Fields");++var_i)
-  {
-    // Determine the variable name 
-    std::string var_name = var_params.get<std::string>(ekat::strint("field",var_i+1));
-    m_fields.push_back(var_name);
-    /* Check that all dimensions for this variable are set to be registered */
+
+  auto avg_type = m_params.get<std::string>("AVERAGING TYPE");
+  m_avg_type = ekat::upper_case(avg_type);
+  auto valid_avg_types = {"INSTANT", "MAX", "MIN", "AVERAGE"};
+  EKAT_REQUIRE_MSG (ekat::contains(valid_avg_types,m_avg_type),
+      "Error! Unsupported averaging type '" + avg_type + "'.\n"
+      "       Valid options: Instant, Max, Min, Average. Case insensitive.\n");
+
+  m_has_restart_data = (m_avg_type!="INSTANT");
+
+  if (m_has_restart_data) {
+    // This avg_type needs to save  some info in order to restart the output.
+    // E.g., we might save 30-day avg value for field F, but due to job size
+    // break the run into three 10-day runs. We then need to save the state of
+    // our averaging in a "restart" file (e.g., the current avg and avg_count).
+    // If checkpoint_freq=0, we do not perform checkpointing of the output.
+    // NOTE: we may not need to *read* restart data (if this is not a restart run),
+    //       but we might still want to save it.
+    m_checkpoint_freq = m_params.get<int>("CHECKPOINT FREQUENCY",0);
+  }
+
+  // For each output field, ensure its dimensions are registered in the pio file
+  m_fields = m_params.get<std::vector<std::string>>("FIELDS");
+  for (const auto& var_name : m_fields) {
     register_dimensions(var_name);
   }
 
   // Now that the fields have been gathered register the local views which will be used to determine output data to be written.
   register_views();
 
-  // If this is a restart run that requires a restart history file read input here:
-  if (m_read_restart_hist)
-  {
+  // If this is "normal" output of a restarted run, and avg_type requires restart data,
+  // then we have to open the history restart file and read its data.
+  // The user can skip this (which means the Output averaging would start from scratch)
+  // by setting "Restart History" to false.
+  m_is_restarted_run &= m_params.get("Restart History",true);
+  if (m_has_restart_data && m_is_restarted_run && !m_is_model_restart_output) {
     std::ifstream rpointer_file;
     rpointer_file.open("rpointer.atm");
     std::string filename;
     bool found = false;
-    std::string testname = m_casename+"."+m_avg_type+"."+m_out_units+"_x"+std::to_string(m_out_frequency);
-    while (rpointer_file >> filename)
-    {
-      if (filename.find(testname) != std::string::npos)
-      {
+    std::string testname;
+    if (m_params.isParameter("RESTART HISTORY FILENAME")) {
```

Comment:
Are all these parameter names written down somewhere?

Reply:
I doubt it. I will add some docs in the output header.